src/main/java/com/atherys/quests/managers/LocationManager.java

```diff
@@ -14,18 +17,66 @@
 
 public final class LocationManager extends AbstractMongoDatabaseManager<LocationManager.QuestLocation> {
 
-    protected LocationManager() {
+    private static LocationManager instance = new LocationManager();
+
+    private Gson gson = GsonUtils.getGson();
+
+    private LocationManager() {
         super(AtherysQuests.getInstance().getLogger(), QuestsDatabase.getInstance(), "questLocations");
     }
 
+    public Optional<QuestLocation> getByLocation(Location<World> location){
+        for(QuestLocation ql : getCache().values()){
+            if(ql.contains(location)){
+                return Optional.of(ql);
+            }
+        }
+        return Optional.empty();
+    }
+
+    public static LocationManager getInstance(){
+        return instance;
+    }
+
+    public void saveAll(){
+       saveAll(getCache().values());
+    }
+
+    public boolean addQuestLocation(Location<World> location, String questId, double radius){
+        Optional<QuestLocation> questLocation = QuestManager.getInstance().getQuest(questId).map(quest ->
+            new QuestLocation(location, quest, radius));
+
+        if(questLocation.isPresent()){
+            QuestLocation questLoc = questLocation.get();
+            for(QuestLocation ql : this.getCache().values()){
+                if(questLoc.overlaps(ql)) return false;
+            }
+            this.save(questLoc);
+            return true;
+        } else return false;
+    }
+
     @Override
     protected Optional<Document> toDocument(QuestLocation questLocation) {
-        return Optional.empty();
+        Document document = new Document();
+
+        Document location = DbUtils.Serialize.location(questLocation.location);
+        document.append("location", location);
+
+        document.append("questId", questLocation.questId);
+        document.append("radius", questLocation.radius);
```

Comment:
Please access these fields using getters. Also, create getters for them. Don't need to make setters, a QuestLocation can be Immutable, that would be just fine, but accessing them directly like this is quite ugly and not very safe.